src/adjoint/masterRoutines.F90

```diff
@@ -94,14 +98,24 @@ subroutine master(useSpatial, famLists, funcValues, forces, &
           call setPointers(nn, 1, sps)
 
           if (useSpatial) then
+
+             call VecGetArrayF90(xSurfVec(1, sps), xSurf, ierr)
+             call EChk(ierr,__FILE__,__LINE__)
+    
```

Comment:
Is this a bug fix? Is there something I need to check here?

Reply:
It is a bug fix. 
But, this part of the master routine is not run during normal operations. 
You can see that to be consistent with the FWD routine, this must be there.